client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -179,6 +180,21 @@ static Request createIndex(CreateIndexRequest createIndexRequest) throws IOExcep
         return new Request(HttpPut.METHOD_NAME, endpoint, parameters.getParams(), entity);
     }
 
+    static Request putMapping(PutMappingRequest putMappingRequest) throws IOException {
+        // The concreteIndex is an internal concept, not applicable to requests made over the REST API.
+        assert (putMappingRequest.getConcreteIndex() == null);
```

Comment:
given that a request is provided by users, this should rather be an exception that gets thrown. Also this assertion may make the user's application crash. Throw `IllegalArgumentException` instead?